Detectors/ITSMFT/MFT/base/include/MFTBase/GeometryTGeo.h

```diff
@@ -9,152 +9,178 @@
 // or submit itself to any jurisdiction.
 
 /// \file GeometryTGeo.h
-/// \brief A simple interface class to TGeoManager
-/// \author bogdan.vulpescu@cern.ch 
-/// \date 01/08/2016
+/// \brief Definition of the GeometryTGeo class
+/// \author bogdan.vulpescu@clermont.in2p3.fr - adapted from ITS, 21.09.2017
+
+#ifndef ALICEO2_MFT_GEOMETRYTGEO_H
+#define ALICEO2_MFT_GEOMETRYTGEO_H
+
+#include <vector>
+#include <array>
+#include <string>
+#include <TGeoMatrix.h> // for TGeoHMatrix
+#include <TObject.h>    // for TObject
+#include "DetectorsBase/DetID.h"
+#include "DetectorsBase/Utils.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "ITSMFTBase/GeometryTGeo.h"
+#include "Rtypes.h" // for Int_t, Double_t, Bool_t, UInt_t, etc
+
+class TGeoPNEntry; 
+namespace o2 {
+namespace ITSMFT {
+  class Segmentation;
+  class SegmentationPixel;
+}
+}
 
-#ifndef ALICEO2_MFT_GEOMETRYTGEO_H_
-#define ALICEO2_MFT_GEOMETRYTGEO_H_
+namespace o2
+{
+  
+namespace MFT
+{
 
-#include <boost/bimap.hpp>
+class GeometryTGeo : public o2::ITSMFT::GeometryTGeo
+{
+ public:
 
-#include "TObject.h"
-#include "TString.h"
-#include "TObjArray.h"
-#include "TGeoMatrix.h"
+  typedef o2::Base::Transform3D Mat3D;
+  using DetMatrixCache::getMatrixT2L;
+  using DetMatrixCache::getMatrixL2G;
+  using DetMatrixCache::getMatrixT2G;
 
-namespace o2 {
-  namespace ITSMFT {
-    class Segmentation;
+  static GeometryTGeo* Instance() {
+    // get (create if needed) a unique instance of the object
+    if (!sInstance) sInstance = std::unique_ptr<GeometryTGeo>(new GeometryTGeo(true, true, 0));
```

Comment:
@bovulpes , @shahor02 : I am wondering if there is any benefit of having a global static variable `sInstance` to implement a singleton? The alternative code would be 
```
static GeometryTGeo* Instance() {
  static GeometryTGeo instance(true,true,0);
  return &instance;
}
``` 
and does not require the static member variable.

Reply:
@sawenzel : the option of initialization from the stored instance would require a static member on the class level, since should be done via different method.

Reply:
I see. You are saying that this is the approach to combine singletons with ROOT persistency. Fine.